poglink/models/rates.py

```diff
@@ -0,0 +1,149 @@
+from dataclasses import dataclass, field
+from typing import Any, List, Tuple
+
+
+class RatesStatus:
+
+    RATES_NAMES = {
+        "TamingSpeedMultiplier": "Taming",
+        "HarvestAmountMultiplier": "Harvesting",
+        "XPMultiplier": "XP",
+        "MatingIntervalMultiplier": "Mating Interval",
+        "BabyMatureSpeedMultiplier": "Maturation",
+        "EggHatchSpeedMultiplier": "Hatching",
+        "BabyCuddleIntervalMultiplier": "Cuddle Interval",
+        "BabyImprintAmountMultiplier": "Imprinting",
+        "HexagonRewardMultiplier": "Hexagon Reward",
+    }
+
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        expected, extras = self.get_expected_and_extras(kwargs)
+
+        for k, v in expected.items():
+            setattr(self, k, v)
+
+        self.extras = extras
+
+    def __eq__(self, __o: object) -> bool:
+        return self.__dict__ == __o.__dict__
+
+    @staticmethod
+    def parse_raw(raw: str) -> dict:
+        parsed_vals = {
+            key.strip(): val.strip()
+            for key, val in [line.split("=") for line in raw.strip().splitlines()]
+        }
+        return parsed_vals
+
+    @staticmethod
+    def get_expected_and_extras(parsed_vals_dict: dict) -> Tuple[dict, dict]:
+        expected = {
+            k: v for k, v in parsed_vals_dict.items() if k in RatesStatus.RATES_NAMES
+        }
+        extras = {
+            k: v
+            for k, v in parsed_vals_dict.items()
+            if k not in RatesStatus.RATES_NAMES and k != "extras"
+        }
+
+        if "extras" in parsed_vals_dict:
+            extras.update(parsed_vals_dict.get("extras"))
+
+        return expected, extras
+
+    @classmethod
+    def from_raw(cls, raw_txt: str):
+        val_dict = cls.parse_raw(raw_txt)
+
+        return cls.from_dict(val_dict)
+
+    @classmethod
+    def from_dict(cls, val_dict):
+        expected, extras = cls.get_expected_and_extras(val_dict)
+
+        return cls(**expected, **extras)
+
+    def update_vals(self, vals, raw=False):
+
+        parsed_vals = self.parse_raw(vals) if raw else vals
+
+        expected, extras = self.get_expected_and_extras(parsed_vals)
+
+        for k, v in expected.items():
+            setattr(self, k, v)
+
+        self.extras.update(extras)
+
+    def to_dict(self):
+        expected, extras = self.get_expected_and_extras(self.__dict__)
+
+        output_dict = expected
+        output_dict.update(extras)
+
+        return output_dict
+
+    def to_raw(self):
+        return "\n".join([f"{k}={v}" for k, v in self.to_dict().items()])
+
+    def get_diff(self, newrates: "RatesStatus"):
+        return RatesDiff.from_statuses(self, newrates)
```

Comment:
This method used to contain the looped construction of the diff object, but I've moved that logic into the `RatesDiff` class to reduce complexity in this class.